ci/jenkins/scale-agent.sh

```diff
@@ -0,0 +1,285 @@
+      "tunnel": "",
+      "vmargs": ""
+   },
+   "retentionStrategy": {
+      "stapler-class": "hudson.slaves.RetentionStrategy$Always",
+      "$class": "hudson.slaves.RetentionStrategy$Always"
+   },
+   "nodeProperties": {
+      "stapler-class-bag": "true",
+      "hudson-slaves-EnvironmentVariablesNodeProperty": {
+         "env": [
+            {
+               "key": "JAVA_HOME",
+               "value": "/usr/lib/jvm/java-11-openjdk-amd64"
+            }
+         ]
+      },
+      "_comment:": {
+         "hudson-tools-ToolLocationNodeProperty": {
+           "locations": [
+               {
+                  "key": "hudson.model.JDK$DescriptorImpl@JAVA-11",
+                  "home": "/usr/bin/java"
+               }
+            ]
+         }
+      }
+   }
+}
+EOF
+    
+    response_code=$(curl -L -s -k -v -w "%{http_code}" -u $JENKINS_USER:$JENKINS_API_TOKEN -H "Content-Type:application/x-www-form-urlencoded" -X POST \
+        -d "json=$(cat node.json)" "$JENKINS_URL/computer/doCreateItem?name=$NEW_AGENT_NAME&type=hudson.slaves.DumbSlave")
+    if [[ "$response_code" -ne 200 ]]; then
+        echoerr "Failed to create agent. HTTP status code: $response_code"
+        return 1
+    fi
+
+    jnlp_url="$JENKINS_USER/computer/$NEW_AGENT_NAME/slave-agent.jnlp"
+    curl -k -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$jnlp_url" -o agent.jnlp
+    secret=$(grep -oP '(?<=<argument>)[^<]+' agent.jnlp | head -1)
+
+    # Run from agent command line
+    gcloud compute ssh "$NEW_AGENT_NAME" --zone="$ZONE" --command \
+        "curl -ksO $JENKINS_URL/jnlpJars/agent.jar && java -jar agent.jar -url $JENKINS_URL -secret $secret -name $NEW_AGENT_NAME"
+    
+    echo "Agent $NEW_AGENT_NAME has been added."
+}
+
+function remove_agent {
+    AGENT_LIST=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/api/json" | jq -r '.computer[] | .displayName')
+
+    if [ -z "$AGENT_LIST" ]; then
+        echoerr "No agents found."
+        return 1
+    fi
+    
+    DELETED_ONCE=false
+    for AGENT_NAME in $AGENT_LIST; do
+        if [[ "$AGENT_NAME" == *"$AGENT_NAME_PATTERN"* ]]; then
+            echo "Checking Agent: $AGENT_NAME"
+
+            RUNNING_JOBS=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/$AGENT_NAME/api/json" | jq '[.executors[] | select(.currentExecutable != null)] | length')
+
+            if [ "$RUNNING_JOBS" -eq 0 ]; then
+                echo "Agent $AGENT_NAME can be safely removed"
+
+                echo "Removing label from agent $AGENT_NAME"
+                curl -X POST "$JENKINS_URL/computer/$AGENT_NAME/label" --user "$JENKINS_USER:$JENKINS_API_TOKEN" -d "labels="
+
+                # Recheck if there are no running jobs
+                RUNNING_JOBS=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/$AGENT_NAME/api/json" | jq '[.executors[] | select(.currentExecutable != null)] | length')
+            
+                if [ "$RUNNING_JOBS" -eq 0 ]; then
```

Comment:
Maybe keep at least 1 agent?

Reply:
I suppose there should be at least one static agent that won't be affected by the scale-up script to handle any unexpected failures. What do you think?

Reply:
OK. So the static agent name will use a different name pattern.